src/notebook/model.ts

```diff
@@ -413,7 +436,7 @@ class NotebookModel implements INotebookModel {
     if (this.activeCellIndex === this.cells.length - 1) {
       let cell = this.createCodeCell();
       this.cells.add(cell);
-      cell.mode = 'edit';  // This already sets the new index.
+      this.mode = 'edit';  // This already sets the new index.
```

Comment:
I am going to explore this part of the code further in the PR that moves `active` and `selected` off of the cells.